src/encoder/mp4_ffmpeg_cli_pipe.rs

```diff
@@ -0,0 +1,276 @@
+        command.arg(&self.path);
+
+        // Set up pipes
+        command
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped());
+
+        let child = command.spawn()?;
+        self.process = Some(child);
+
+        Ok(())
+    }
+
+    /// Converts Bevy Image (RGBA) to raw bytes suitable for ffmpeg
+    fn image_to_raw_bytes(image: &Image) -> Result<Vec<u8>> {
+        // Bevy images are typically in RGBA format
+        let image_data = image.data.as_ref().ok_or("Image has no data")?;
+
+        // ffmpeg expects raw RGBA bytes
+        Ok(image_data.clone())
+    }
+}
+
+impl Encoder for Mp4FfmpegCliPipeEncoder {
+    fn encode(&mut self, image: &Image) -> Result<()> {
+        // Initialize process on first frame
+        if self.process.is_none() {
+            self.initialize_process(image)?;
+        }
+
+        // Get the stdin pipe
+        let process = self
+            .process
+            .as_mut()
+            .ok_or("FFmpeg process not initialized")?;
+
+        let stdin = process.stdin.as_mut().ok_or("Failed to get stdin pipe")?;
+
+        // Convert image to raw bytes and write to ffmpeg
+        let raw_bytes = Self::image_to_raw_bytes(image)?;
+        stdin.write_all(&raw_bytes)?;
+        stdin.flush()?;
+
+        Ok(())
+    }
+
+    fn finish(mut self: Box<Self>) {
+        if let Some(mut process) = self.process.take() {
+            // Close stdin to signal end of input
+            drop(process.stdin.take());
+
+            // Wait for process to complete
+            match process.wait() {
+                Ok(status) => {
+                    if status.success() {
+                        bevy::log::info!("Video encoding completed successfully: {:?}", self.path);
+                    } else {
+                        // Try to capture stderr for error details
+                        if let Some(mut stderr) = process.stderr.take() {
+                            use std::io::Read;
+                            let mut error_msg = String::new();
+                            if stderr.read_to_string(&mut error_msg).is_ok() {
+                                bevy::log::error!(
+                                    "FFmpeg failed with status {}: {}",
+                                    status,
+                                    error_msg
+                                );
+                            } else {
+                                bevy::log::error!("FFmpeg failed with status: {}", status);
+                            }
+                        } else {
+                            bevy::log::error!("FFmpeg failed with status: {}", status);
+                        }
+                    }
+                }
+                Err(error) => {
+                    bevy::log::error!("Failed to wait for FFmpeg process: {:?}", error);
+                }
+            }
+        }
+    }
+}
+
+impl Drop for Mp4FfmpegCliPipeEncoder {
```

Comment:
This seems kind of hacky with the sleep, but also not sure if there is a better way?